include/circt/Dialect/Seq/SeqOps.td

```diff
@@ -350,6 +350,28 @@ def ClockMuxOp : SeqOp<"clock_mux", [Pure]> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Clock Dividers
+//===----------------------------------------------------------------------===//
+
+def ClockDivider2 : SeqOp<"clock_div_2", [Pure]> {
+  let summary = "Produces a clock divided by 2";
```

Comment:
Could we make the division factor an attribute of the op?

Reply:
Yes, I was thinking that this is awfully specific.

Reply:
But it should be restricted to a power of 2. I suspect anything else would fuck too much with the harmonics of the clock tree so I would think smart designers would avoid it...

Reply:
Question is if you want that very niche concern to restrict the entire IR. As you say, designers could just avoid it. But with PLLs running clocks at totally awkward frequencies in the GHz range, which is likely producing a lot more problems, we probably don't want to deal with this in the IR.

But maybe forcing the division to be a *multiple* of two is a good idea, for a 50% duty cycle? Or maybe we just don't restrict this at all in the IR and let the users deal with it.